scene/main/viewport.cpp

```diff
@@ -1457,17 +1457,24 @@ void Viewport::_gui_show_tooltip() {
 		return;
 	}
 
+	// Controls can implement `make_custom_tooltip` to provide their own tooltip.
+	// This should be a Control node which will be added as child to a TooltipPanel.
+	Control *base_tooltip = tooltip_owner->make_custom_tooltip(gui.tooltip_text);
+
+	// When the custom control is not visible, don't show any tooltip.
+	// This way, the custom tooltip from ConnectionsDockTree can create
+	// its own tooltip without conflicting with the default one, even an empty tooltip.
+	if (base_tooltip && !base_tooltip->is_visible()) {
+		return;
+	}
```

Comment:
This is a nice workaround! :+1: Instead of `is_visible()`, we could check a method or metadata to ensure compatibility is preserved, but the current option looks good to me too.

Reply:
I'll let it like that then, thanks.